dockstore-client/src/test/java/io/dockstore/client/cli/LaunchTestIT.java

```diff
@@ -1190,6 +1198,93 @@ public void cwlMissingIdParameters() {
         assertTrue("output should include an error message", systemErrRule.getLog().contains("Syntax error while parsing a block collection"));
     }
 
+    @Test
+    public void entry2jsonNoVersion() {
+        /*
+         * Make a runtime JSON template for input to the workflow
+         * but don't provide a version at the end of the entry
+         * E.g dockstore workflow convert entry2json --entry quay.io/collaboratory/dockstore-tool-linux-sort
+         * Dockstore will try to use the 'master' version, however the 'master' version
+         * is not valid so Dockstore should print an error message and exit
+         * */
+        WorkflowVersion aWorkflowVersion1 = new WorkflowVersion();
+        aWorkflowVersion1.setName("master");
+        aWorkflowVersion1.setValid(false);
+        Date earlierDate = new Date(100L);
+        aWorkflowVersion1.setLastModified(earlierDate);
+
+        List<WorkflowVersion> listWorkflowVersions = new ArrayList<WorkflowVersion>();
+        listWorkflowVersions.add(aWorkflowVersion1);
+
+        Workflow workflow = new Workflow();
+        workflow.setWorkflowVersions(listWorkflowVersions);
+        workflow.setLastModified(1);
+
+        WorkflowsApi api = mock(WorkflowsApi.class);
+        UsersApi usersApi = mock(UsersApi.class);
+        Client client = new Client();
+
+        doReturn(workflow).when(api).getPublishedWorkflowByPath(anyString(), eq(null));
+
+        WorkflowClient workflowClient = new WorkflowClient(api, usersApi, client, false);
+
+        exit.expectSystemExit();
+        exit.checkAssertionAfterwards(
+                () -> assertTrue("output should include error message",
+                        systemErrRule.getLog().contains("Cannot use workflow version 'master'")));
+
+        try {
+            workflowClient.downloadTargetEntry("quay.io/collaboratory/dockstore-tool-linux-sort", ToolDescriptor.TypeEnum.WDL, false);
+        } catch (IOException e) {
+
```

Comment:
This empty catch is confusing, and I thought Checkstyle caught these, but maybe it doesn't apply to tests.

Does your test throw an IOException? If it does, could you please add a comment in the block. If it doesn't throw an IOException, I would remove the try...catch entirely, and have the test method throw an IOException.

The issue with the code as it is now, assuming you don't throw an IOException

1. Test passes today
2. Somebody in the future introduces a bug that causes the method to throw an IOException when it shouldn't.
3. Your test would continue to pass, because you're silently ignoring the exception.

If you're throwing an exception in the test, then that's different, but I think you should at least have a comment saying that it's expected.

Reply:
Now throw IOException up the call stack so I don't have to catch it. Removed the try catch block